File: Resource.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace MiddleAges
{
    class Resource
    {
        public Resource(int _gold, int _wood, int _stone, int _iron, int _food)
        {
            GoldQuantity = _gold;
            WoodQuantity = _wood;
            StoneQuantity = _stone;
            IronQuantity = _iron;
            FoodQuantity = _food;
        }

        //#################################################
        // MEMBER FUNCTIONS
        //#################################################
        // Remove
        public void RemoveGold(int quantity)
        {
            _goldQuantity -= quantity;
        }
        public void RemoveWood(int quantity)
        {
            _woodQuantity -= quantity;
        }
        public void RemoveStone(int quantity)
        {
            _stoneQuantity -= quantity;
        }
        public void RemoveIron(int quantity)
        {
            _ironQuantity -= quantity;
        }
        public void RemoveFood(int quantity)
        {
            _foodQuantity -= quantity;
        }
    
        // Add 
        public void AddGold(int quantity)
        {
            _goldQuantity += quantity;
        }
        public void AddWood(int quantity)
        {
            _woodQuantity += quantity;
        }
        public void AddStone(int quantity)
        {
            _stoneQuantity += quantity;
        }
        public void AddIron(int quantity)
        {
            _ironQuantity += quantity;
        }
        public void AddFood(int quantity)
        {
            _foodQuantity += quantity;
        }

        //#################################################
        // SETTERS & GETTERS
        //#################################################

        public int GoldQuantity
        {
            get
            {
                return _goldQuantity;
            }
            set
            {
                _goldQuantity = value;
            }
        }

        public int WoodQuantity
        {
            get
            {
                return _woodQuantity;
            }
            set
            {
                _woodQuantity = value;
            }
        }

        public int StoneQuantity
        {
            get
            {
                return _stoneQuantity;
            }
            set
            {
                _stoneQuantity = value;
            }
        }

        public int IronQuantity
        {
            get
            {
                return _ironQuantity;
            }
            set
            {
                _ironQuantity = value;
            }
        }

        public int FoodQuantity
        {
            get
            {
                return _foodQuantity;
            }
            set
            {
                _foodQuantity = value;
            }
        }

        //#################################################
        // MEMBER VARIABLES
        //#################################################
        private int _goldQuantity;
        private int _woodQuantity;
        private int _stoneQuantity;
        private int _ironQuantity;
        private int _foodQuantity;
    }
}

File: Etap 4 - Implementacja C++/Implementacja_C++/MiddleAges/resource.cpp
#include "resource.h"

//#################################################
// CONSTRUCTORS
//#################################################
Resource::Resource()
{
}

Resource::Resource(int gold, int wood, int stone, int iron, int food)
{
    setGoldQuantity(gold);
    setWoodQuantity(wood);
    setStoneQuantity(stone);
    setIronQuantity(iron);
    setFoodQuantity(food);
}

//#################################################
// MEMBER FUNCTIONS
//#################################################
// remove
void Resource::removeGold(int quantity)
{
    goldQuantity -= quantity;
}

void Resource::removeWood(int quantity)
{
    woodQuantity -= quantity;
}

void Resource::removeStone(int quantity)
{
    stoneQuantity -= quantity;
}

void Resource::removeIron(int quantity)
{
    ironQuantity -= quantity;
}

void Resource::removeFood(int quantity)
{
    foodQuantity -= quantity;
}

// add
void Resource::addGold(int quantity)
{
    goldQuantity += quantity;
}

void Resource::addWood(int quantity)
{
    woodQuantity += quantity;
}

void Resource::addStone(int quantity)
{
    stoneQuantity += quantity;
}

void Resource::addIron(int quantity)
{
    ironQuantity += quantity;
}

void Resource::addFood(int quantity)
{
    foodQuantity += quantity;
}

//#################################################
// SETTERS
//#################################################
void Resource::setGoldQuantity(int gold)
{
    goldQuantity = gold;
}

void Resource::setWoodQuantity(int wood)
{
    woodQuantity = wood;
}

void Resource::setStoneQuantity(int stone)
{
    stoneQuantity = stone;
}

void Resource::setIronQuantity(int iron)
{
    ironQuantity = iron;
}

void Resource::setFoodQuantity(int food)
{
    foodQuantity = food;
}

//#################################################
// GETTERS
//#################################################
int Resource::getGoldQuantity()
{
    return goldQuantity;
}
int Resource::getWoodQuantity()
{
    return woodQuantity;
}
int Resource::getStoneQuantity()
{
    return stoneQuantity;
}
int Resource::getIronQuantity()
{
    return ironQuantity;
}
int Resource::getFoodQuantity()
{
    return foodQuantity;
}
